libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Preds/CertState.hs

```diff
@@ -47,14 +47,20 @@ manyCoin size = do
 -- ======================================
 
 vstatePreds :: Era era => Proof era -> [Pred era]
-vstatePreds _p =
+vstatePreds p = vstateGenPreds p ++ vstateCheckPreds p
+
+vstateGenPreds :: Era era => Proof era -> [Pred era]
+vstateGenPreds _p =
```

Comment:
Which of these are generation constraints and which are checking constraints?

Reply:
I guess
these two
 [ Sized (Range 0 0) currentDRepState
    , Sized (Range 0 0) committeeState
    ]
are checking, since they need to be empty before Conway Era

There is some checking constrain, implied by this comment
... he deposit field of the currentDRepState is the drepDeposit
But I am not sure how to best express this. It goes something like this. Add up all the deposit fields of the 'currentDRepState' this must be less then the utxosDeposit field in the utxoState. It is more prcise than that, since there are 4 places deposits are stored, and the sum of all 4 must be equal to the variable 'deposits'

Reply:
I've done this but I think we'll leave the details of implementing the constraints for the next PR - maybe after we've gotten some more input from the team on the constraints.